File: 2022-2023 ICPC Brazil Subregional Programming Contest/D.py
# https://codeforces.com/gym/103960/problem/D

def solve(n: int, x: int, y: int) -> int:
    res: int = 0
    while x != 2**(n-1):
        if x < 2**(n-1):
            res += 1
            x = 2*x
        else:
            x = 2**n - x
    return res


def main():
    n, x, y = map(int, input().split(' '))
    print(solve(n, x, y))


if __name__ == '__main__':
    main()

File: 2022-2023 ICPC Brazil Subregional Programming Contest/D.cpp
// https://codeforces.com/gym/103960/problem/D

#include <iostream>

// using namespace std;

unsigned int solve(unsigned int N, unsigned int x)
{
    unsigned int res = 0;
    while (x != 1 << (N - 1))
    {
        if (x < 1 << (N - 1))
        {
            res++;
            x = x * 2;
        }
        else
        {
            x = (1 << N) - x;
        }
    }
    return res;
}

int main(void)
{
    // Parse N, x, y from stdin
    unsigned int N, x;
    std::cin >> N >> x;

    unsigned int res = solve(N, x);

    std::cout << res;

    return EXIT_SUCCESS;
}
